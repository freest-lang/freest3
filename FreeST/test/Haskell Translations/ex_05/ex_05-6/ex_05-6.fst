-- V exercise 6

data IntList = End | List Int IntList

dropWhile' : (Int -> Bool) -> IntList -> IntList
dropWhile' f list =
    case list of {
        End -> End,
        List x rest -> 
            if f x 
                then dropWhile' f rest
                else List x rest
    }

isEven : Int -> Bool
isEven x = mod x 2 == 0

list : IntList
list = List 4 (List 2 (List 6 (List 1 (List 8 (List 6 (List 2 End))))))

main : IntList
main = dropWhile' isEven list
-- result = List 1 (List 8 (List 6 (List 2 End)))
