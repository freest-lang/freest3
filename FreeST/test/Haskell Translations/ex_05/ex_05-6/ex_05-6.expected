List 1 (List 8 (List 6 (List 2 End)))